src/coprocessor/Client.cc

```diff
@@ -461,15 +460,56 @@ std::vector<BatchCopTask> buildBatchCopTasks(
         // store_addr -> BatchCopTask
         std::map<std::string, BatchCopTask> store_task_map;
         bool need_retry = false;
+
+        std::map<uint64_t, kv::Store> alive_tiflash_stores;
+        if (filter_alive_tiflash_stores)
+        {
+            bool has_force_get_all_stores = false;
+            while (alive_tiflash_stores.empty())
+            {
+                // TODO: Need find a way to drop store info in RegionCache. Otherwise we may have meanless probing of WN.
+                // We don't directly drop the cached store info in RegionCache here, because the current mechanism of RegionCache is to
+                // cache store information by looking up the store corresponding to region.peer from PD.
+                // Therefore, the removal mechanism should also be consistent: only when all cached regions on a store have been dropped should the
+                // cached store info be removed. Otherwise, it may lead to inconsistencies between RegionCache::region and RegionCache::store.
+                auto tiflash_stores = cache->getAllTiFlashStores(label_filter, /*exclude_tombstone =*/true);
+                log->information(stores_to_str("before filter alive stores: ", tiflash_stores));
+                // filter by the mpp_prober
+                alive_tiflash_stores = details::filterAliveStores(cluster, tiflash_stores, log);
+                log->information(stores_to_str("after filter alive stores: ", alive_tiflash_stores));
+                if (alive_tiflash_stores.empty())
+                {
+                    if (!has_force_get_all_stores)
+                    {
+                        // Get all stores immediately first time.
+                        log->warning("no alive tiflash, cannot dispatch BatchCopTask, force get all stores from pd", ErrorCodes::CoprocessorError);
+                        has_force_get_all_stores = true;
+                        cache->forceReloadAllStores();
+                    }
+                    else
+                    {
+                        // Backoff and get all stores.
+                        log->warning("no alive tiflash, cannot dispatch BatchCopTask, retrying", ErrorCodes::CoprocessorError);
+                        bo.backoff(kv::boTiFlashRPC, Exception("No alive tiflash WN"));
```

Comment:
```suggestion
                        bo.backoff(kv::boTiFlashRPC, Exception("No alive tiflash stores"));
```
I think this logic does not only apply to disagg arch, but also the non-disagg arch. So using "tiflash stores" instead of "tiflash WN"